src/main/scala/io/qbeast/spark/table/IndexedTable.scala

```diff
@@ -477,13 +493,29 @@ private[table] class IndexedTableImpl(
 
   }
 
-  override def optimize(revisionID: RevisionID, options: Map[String, String]): Unit = {
+  override def optimize(
+      revisionID: RevisionID,
+      fraction: Double,
+      options: Map[String, String]): Unit = {
+    assert(fraction > 0d && fraction <= 1d)
     val indexFiles = snapshot.loadIndexFiles(revisionID)
     import indexFiles.sparkSession.implicits._
-    val files = snapshot.loadIndexFiles(revisionID).map(_.path).as[String].collect()
+    val files = indexFiles.transform(filterSamplingFiles(fraction)).map(_.path).collect()
     optimize(files, options)
   }
 
+  private[table] def filterSamplingFiles(
```

Comment:
Why don't we use a method in  the`SamplingStrategy` to read the files?

Reply:
It is a bit tricky because there is no way to specify the `revisionID` in `SamplingStrategy.`

Reply:
oh, okei, we can leave that for another PR